src/EnergyPlus/IceThermalStorage.cc

```diff
@@ -97,361 +96,207 @@ namespace IceThermalStorage {
-        IceStorage.deallocate();
-        IceStorageReport.deallocate();
-        DetIceStor.deallocate();
-        IceStorageTypeMap.deallocate();
+        NumSimpleIceStorage = 0;
+        NumDetailedIceStorage = 0;
+        TotalNumIceStorage = 0;
+        SimpleIceStorage.deallocate();
+        DetailedIceStorage.deallocate();
     }
 
-    void SimIceStorage(std::string const &IceStorageType,
-                       std::string const &IceStorageName,
-                       int &CompIndex,
-                       bool const RunFlag,
-                       bool const FirstIteration,
-                       bool const InitLoopEquip,
-                       Real64 &MyLoad)
+    PlantComponent *SimpleIceStorageData::factory(std::string const &objectName)
     {
-
-        // SUBROUTINE INFORMATION:
-        //       AUTHOR
-        //       DATE WRITTEN
-        //       MODIFIED       na
-        //       RE-ENGINEERED  na
-
-        // PURPOSE OF THIS SUBROUTINE:
-
-        // Using/Aliasing
-        using DataGlobals::BeginEnvrnFlag;
-        using DataPlant::DualSetPointDeadBand;
-        using DataPlant::PlantLoop;
-        using DataPlant::SingleSetPoint;
-        using FluidProperties::GetSpecificHeatGlycol;
-        using ScheduleManager::GetCurrentScheduleValue;
-
-        // Locals
-        // SUBROUTINE ARGUMENT DEFINITIONS:
-        Real64 DemandMdot;
-        Real64 TempIn;
-        Real64 TempSetPt(0.0);
-        Real64 MyLoad2;
-        Real64 MaxCap;
-        Real64 MinCap;
-        Real64 OptCap;
-        Real64 Cp; // local plant fluid specific heat
-
-        // SUBROUTINE PARAMETER DEFINITIONS:
-        static std::string const RoutineName("SimIceStorage");
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        static bool firstTime(true);
-        static bool MyEnvrnFlag(true);
-        int IceStorageNum;
-
-        // FLOW
-
-        //  Set initialization flags
-        //  Allow ice to build up during warmup?
-        //  IF( (BeginEnvrnFlag) .OR. (WarmupFlag) ) THEN
-        if (BeginEnvrnFlag && MyEnvrnFlag) {
-            ResetXForITSFlag = true;
-            MyEnvrnFlag = false;
+        // Process the input data for boilers if it hasn't been done already
+        if (getITSInput) {
+            GetIceStorageInput();
+            getITSInput = false;
         }
 
-        if (!BeginEnvrnFlag) {
-            MyEnvrnFlag = true;
+        // Now look for this particular pipe in the list
+        for (auto &ITS : SimpleIceStorage) {
+            if (ITS.Name == objectName) {
+                return &ITS;
+            }
         }
+        // If we didn't find it, fatal
+        ShowFatalError("LocalSimpleIceStorageFactory: Error getting inputs for simple ice storage named: " + objectName); // LCOV_EXCL_LINE
+        // Shut up the compiler
+        return nullptr; // LCOV_EXCL_LINE
+    }
 
-        if (firstTime) {
+    PlantComponent *DetailedIceStorageData::factory(std::string const &objectName)
```

Comment:
Interesting.  I didn't actually realize there were two totally separate objects in here.  No issues for now, though I wonder if the overall module could be cleaned up later to just use one object with small variations in terms of physics calculations.  This all looks fine though.

Reply:
Agreed. I'm not sure when the right time is to do these types of cleanups, but I haven't been worrying about it too much on the plant-comp refactor work.